### PR TITLE
Upgrade Vue.js to v3 and drop IE11 support

### PR DESCRIPTION
**Pull request by mvorisek**
Major version upgrades:
```
css-loader                        ^5.2.6  →    ^6.7.1
mitt                              ^2.1.0  →    ^3.0.0
style-loader                      ^2.0.0  →    ^3.3.1
vue                              ^2.6.14  →   ^3.2.41
vue-flatpickr-component           ^8.1.7  →   ^10.0.0
vue-loader                       ^15.9.7  →   ^17.0.0
vue-template-compiler            ^2.7.13  →   removed
vue-flatpickr-component           ^8.1.7  →   ^10.0.0
semantic-ui-vue                  ^0.11.0  →   removed
vue-fomantic-ui                    added  →   ^0.13.0
vue-flatpickr-component           ^8.1.7  →   ^10.0.0
vue-query-builder                 ^0.8.2  →   github:TheAustinG/vue-query-builder#2f5c4408e3
```

- css-loader, style-loader
  no change required
- mitt
  no change required
- vue, vue-loader
  https://blog.logrocket.com/refactoring-vue-2-apps-vue-3/
  --> BC break: you must update your components to vue3
- vue-template-compiler
  https://github.com/vuejs/core/tree/main/packages/compiler-sfc
  no change required (integrated into vue package as of 3.2.13)
- semantic-ui-vue/vue-fomantic-ui
  https://github.com/Semantic-UI-Vue/Semantic-UI-Vue/issues/489
  --> BC break: replaced with https://github.com/nightswinger/vue-fomantic-ui
- vue-flatpickr-component
  https://github.com/ankurk91/vue-flatpickr-component/blob/master/UPGRADING.md
  no user change required
- vue-query-builder
  replaced with https://github.com/dabernathy89/vue-query-builder/compare/master...TheAustinG:vue-query-builder:master fork
  no user change required

multiline/vue dropdown API filtering to be fixed later - https://github.com/atk4/ui/issues/1940

BC break: we also drop IE11 support officially (as no longer supported by vue3 - https://stackoverflow.com/questions/64836337/using-vue-3-in-ie11#64837153)

### CODE REVIEW

**Comment by mvorisek** [comment]
Hi @ibelar, would you be please so kind and look at the vue3 upgrade?

**Comment by ibelar** [comment]
> Hi @ibelar, would you be please so kind and look at the vue3 upgrade?

Upgrading to Vue 3 will not be possible until Semantic-Ui vue is upgrade as well or completely refactoring component that use it.

**Comment by mvorisek** [comment]
image

the `semantic-ui-vue` package is used only for multiline component to render table/rows and for dropdown component to render dropdown, maybe we can drop `semantic-ui-vue` package completely

**Comment by mvorisek** [comment]
@ibelar can you please look into the vue scope builder? I found a scope builder fork for vue3, the `demos/form-control/scope-builder.php` demo renders in browser, but no no rule can be added/removed

here is a diff of the lib (original  vs. the new) https://github.com/dabernathy89/vue-query-builder/compare/master...TheAustinG:vue-query-builder:master

File: demos/form-control/multiline.php [hunk 1]
```diff
@@ -29,9 +29,14 @@
 
 // Add multiline field and set model.
 /** @var Form\Control\Multiline */
-$multiline = $form->addControl('ml', [Form\Control\Multiline::class, 'tableProps' => ['color' => 'blue'], 'itemLimit' => 10, 'addOnTab' => true]);
+$multiline = $form->addControl('items', [Form\Control\Multiline::class, 'tableProps' => ['color' => 'blue'], 'itemLimit' => 10, 'addOnTab' => true]);
 $multiline->setModel($inventory);
 
+// DEBUG
```
Comment (mvorisek) [inline demos/form-control/multiline.php:35]: TODO remove before merge

File: demos/javascript/vue-component.php [hunk 2]
```diff
@@ -75,75 +76,98 @@
...
-$clock_template = new HtmlTemplate(<<<'EOF'
+$clockTemplate = new HtmlTemplate(<<<'EOF'
     <div id="{$_id}" class="ui center aligned segment">
-    <my-clock inline-template v-bind="initData">
-        <div>
-            <clock :color="color" :border="border" :bg="bg"></clock>
-            <div class="ui basic segment inline"><div class="ui button primary" @click="onChangeStyle">Change Style</div></div>
-        </div>
-    </my-clock>
+        <my-clock v-bind="initData"></my-clock>
```
Comment (mvorisek) [inline demos/javascript/vue-component.php:108]: TODO or support template in DOM?

File: demos/javascript/vue-component.php [hunk 3]
```diff
@@ -75,75 +76,98 @@
...
-        atk.eventBus.on(this.$root.$el.id + '-clock-change-style', (payload) => {
-            this.onChangeStyle();
-        });
-      },
-      computed: {
-        color: function () {
-          return this.style[this.currentIdx].color
+$clockScript = $app->getTag('script', [], <<<'EOF'
+    // register external clock component
+    // TODO atk.vueService.getVue().component('demo-clock', window.vueDemoClock);
```
Comment (mvorisek) [inline demos/javascript/vue-component.php:116]: TODO remove before merge

File: js/src/services/vue.service.js [hunk 4]
```diff
@@ -62,21 +26,83 @@ class VueService {
...
+
     /**
      * Created a Vue component and add it to the vues array.
      * For root component (App) to be aware that it's children component is
      * mounted, you need to use @hook:mounted="setReady"
      */
-    createAtkVue(id, component, data) {
+    createAtkVue(id, componentName, data) {
+        const app = this.createApp({
+            el: id, // TODO is it needed with mount?
```
Comment (mvorisek) [inline js/src/services/vue.service.js:94]: TODO remove before merge

File: js/src/services/vue.service.js [hunk 5]
```diff
@@ -85,15 +111,30 @@ class VueService {
      * Create a Vue instance from an external src component definition.
      */
     createVue(id, componentName, component, data) {
+        const app = this.createApp({
+            el: id, // TODO is it needed with mount?
```
Comment (mvorisek) [inline js/src/services/vue.service.js:115]: same todo as above

File: js/src/services/vue.service.js [hunk 6]
```diff
@@ -85,15 +111,30 @@ class VueService {
...
+            data: () => ({ initData: data, isReady: true }),
+            mixins: [this.vueMixins],
+        });
+        this._setupComponentAutoloader(app);
+
+        app.component('DemoClock', window.vueDemoClock); // TODO
```
Comment (mvorisek) [inline js/src/services/vue.service.js:121]: TODO remove before merge

File: js/src/services/vue.service.js [hunk 7]
```diff
@@ -103,51 +144,25 @@ class VueService {
...
     registerComponent(component) {
-        // check if that component is already registered
-        const registered = this.vues.filter((comp) => comp.name === component.name);
-        if (registered.length > 0) {
-            registered[0].ids.push(component.ids[0]);
+        if (this.vues[component.name] === undefined) {
+            this.vues[component.name] = component;
         } else {
-            this.vues.push(component);
+            // TODO this.vues[component.name].ids.push(component.ids[0]);
```
Comment (mvorisek) [inline js/src/services/vue.service.js:150]: TODO remove before merge

File: src/Behat/Context.php [hunk 8]
```diff
@@ -582,7 +596,7 @@ public function scopeBuilderReferenceRule(string $name, string $operator, string
 
         $rule = $this->getScopeBuilderRuleElem($name);
         $this->assertSelectedValue($rule, $operator, '.vqb-rule-operator select');
-        $this->assertDropdownValue($rule, $value, '.vqb-rule-input .active.item');
+        // TODO $this->assertDropdownValue($rule, $value, '.vqb-rule-input .active.item');
```
Comment (mvorisek) [inline src/Behat/Context.php:599]: TODO remove before merge

**Comment by mvorisek** [comment]
@ibelar I have managed to upgrade Vue to v3, but I have to admit it was not easy - how much do your apps use custom vue? Would you be ok to have this PR in the next major stable atk4/ui release, or should we release v4.0 first and then tag 4.1 for easier migration?

**Comment by mkrecek234** [comment]
@mvorisek Great, thank you! Even though I am not using a lot of Vue components (only very few) I see that a lot of Vue libraries out there are still in the process of being migrated to v3 - I personally would opt for the 4.1 release for Vue v3 to make upgrade paths more flexible. Too many great improvements beyond Vue v3 so far in Atk4 to bundle them in one release.

File: demos/javascript/vue-component.php [hunk 9]
```diff
@@ -75,75 +76,98 @@
...
+        props: { styles: Array },
+        data: function () {
+            return { style: this.styles, currentIndex: 0 };
         },
-        border: function () {
-          return this.style[this.currentIdx].border
+        mounted: function () {
+            // add a listener for changing clock style.
+            // this will listen to event '-clock-change-style' emit on the eventBus.
+            atk.eventBus.on(this.$root.$el.parentElement.id + '-clock-change-style', (payload) => {
```
Comment (mvorisek) [inline demos/javascript/vue-component.php:131]: TODO investigate why original `this.$root.$el.id` is not working

File: src/App.php [hunk 10]
```diff
@@ -480,7 +480,7 @@ public function initLayout($seed)
     public function initIncludes(): void
     {
         /** @var bool */
-        $minified = true;
+        $minified = false;
```
Comment (mvorisek) [inline src/App.php:483]: TODO before merge

**Pull request merged**